### ISSUE TITLE
Bug: Clicking the `next` button on search results page resets to main index instead of paginating correctly

### ISSUE THREAD

**Issue by TheAnachronism** [ArchiveBox/ArchiveBox#995]
#### Describe the bug

When using the search functionality and getting so many results back that multiple pages are available, using the `next` link at the bottom, resets the search and just goes to the next page of all archived sites. The pagination for search results is available if someone writes the URL correctly themselves, but the links seem broken.

#### Steps to reproduce

- Archive enough sites that multiple pages are available.
- Use the search bar to search for something
- Use the `next` button at the bottom of the page to to got page 2

#### Screenshots or log output

As seen below, first `/public/?q=test` gets called and then only the `page=2` query parameter get set. The combination of the two `/public/?q=test&page=2` works, but the links don't seem to do that correctly.

```logs
"GET /admin/ HTTP/1.1" 200 16927
"GET /admin/ HTTP/1.1" 200 16927
"GET /admin/ HTTP/1.1" 200 16927
"GET /public/ HTTP/1.1" 200 126301
"GET /docs/ HTTP/1.1" 302 0
"GET /public/ HTTP/1.1" 200 126301
"GET /public/?q=test HTTP/1.1" 200 126301
"GET /public/?page=2 HTTP/1.1" 200 126663
```

#### ArchiveBox version

```logs
ArchiveBox v0.6.3
Cpython Linux Linux-5.10.0-11-amd64-x86_64-with-glibc2.31 x86_64
IN_DOCKER=True DEBUG=False IS_TTY=True TZ=UTC SEARCH_BACKEND_ENGINE=ripgrep

[i] Dependency versions:
 √  ARCHIVEBOX_BINARY     v0.6.3          valid     /usr/local/bin/archivebox                                                   
 √  PYTHON_BINARY         v3.10.4         valid     /usr/local/bin/python3.10                                                   
 √  DJANGO_BINARY         v3.1.14         valid     /usr/local/lib/python3.10/site-packages/django/bin/django-admin.py          
 √  CURL_BINARY           v7.74.0         valid     /usr/bin/curl                                                               
 -  WGET_BINARY           -               disabled  /usr/bin/wget                                                               
 √  NODE_BINARY           v17.9.0         valid     /usr/bin/node                                                               
 √  SINGLEFILE_BINARY     v0.3.16         valid     /node/node_modules/single-file/cli/single-file                              
 -  READABILITY_BINARY    -               disabled  /node/node_modules/readability-extractor/readability-extractor              
 -  MERCURY_BINARY        -               disabled  /node/node_modules/@postlight/mercury-parser/cli.js                         
 -  GIT_BINARY            -               disabled  /usr/bin/git                                                                
 -  YOUTUBEDL_BINARY      -               disabled  /usr/local/bin/yt-dlp                                                       
 √  CHROME_BINARY         v100.0.4896.127  valid     /usr/bin/chromium                                                           
 √  RIPGREP_BINARY        v12.1.1         valid     /usr/bin/rg                                                                 

[i] Source-code locations:
 √  PACKAGE_DIR           24 files        valid     /app/archivebox                                                             
 √  TEMPLATES_DIR         4 files         valid     /app/archivebox/templates                                                   
 -  CUSTOM_TEMPLATES_DIR  -               disabled                                                                              

[i] Secrets locations:
 -  CHROME_USER_DATA_DIR  -               disabled                                                                              
 -  COOKIES_FILE          -               disabled                                                                              

[i] Data locations:
 √  OUTPUT_DIR            6 files         valid     /data                                                                       
 √  SOURCES_DIR           739 files       valid     ./sources                                                                   
 √  LOGS_DIR              1 files         valid     ./logs                                                                      
 √  ARCHIVE_DIR           983 files       valid     ./archive                                                                   
 √  CONFIG_FILE           81.0 Bytes      valid     ./ArchiveBox.conf                                                           
 √  SQL_INDEX             4.0 MB          valid     ./index.sqlite3
```

**Comment by EsEnZeT** [comment]
I confirm I'm experiencing exact same issues with latest builds.

**Comment by pirate** [comment]
This should be fixed in the latest versions >v0.7.2 https://github.com/adasneves127/ArchiveBox/pull/2

Let me know if you're still having issues and I'll reopen the ticket.

**Comment by sclu1034** [comment]
I just pulled the latest `:dev` image and I'm running into this issue.

**Comment by pirate** [comment]
Yup I just confirmed I'm seeing the issue too, thanks for reporting.

**Comment by pirate** [comment]
This is fixed as of [v0.8.5-rc](https://github.com/ArchiveBox/ArchiveBox/releases/tag/v0.8.5-rc): https://github.com/ArchiveBox/ArchiveBox/commit/89a066da0bd3631f163781fc8ff549d3c4d947cd
